ccgains/bags.py

```diff
@@ -636,15 +643,53 @@ def deposit(self, dtime, currency, amount, fee, exchange):
             log.info("Taxable loss due to fees: %.3f %s",
                      prof, self.currency)
 
+    def pick_bag(self, exchange, currency):
+        """Pick bag from bag queue according to mode
+
+        :param exchange: (string) The unique name of the
+            exchange/wallet where the funds that are being spent are
+            taken from.
+        :param currency: (string) The currency being spent.
+        """
+        def pick_first():
+            i = 0
+            # next bag in line with fitting currency:
+            while self.bags[exchange][i].currency != currency:
+                i += 1
+                if i == len(self.bags[exchange]):
+                    # Corrupt data error: don't dump state.
+                    raise Exception(
+                        "There are no bags left with the requested currency")
+            return i, self.bags[exchange][i]
+
+        if self.mode == 'FIFO':
+            return pick_first()
+        elif self.mode == 'LIFO':
+            # Iterate bags from the end
+            i = len(self.bags[exchange]) - 1
+            while self.bags[exchange][i].currency != currency:
+                i -= 1
+                if not i:
+                    raise Exception(
+                        "There are no bags left with the requested currency")
+            return i, self.bags[exchange][i]
+        elif self.mode == 'LPFO':
+            # Sort bags by price, lowest first, than pick first with needed currency
+            self.bags[exchange] = sorted(self.bags[exchange], key=lambda bag: bag.price)
```

Comment:
optimization question: Do you guys have any experience with sorting speed? If the bags are already sorted from the first call, sorting them again will be quite fast, right? Then we could leave it as it is. Otherwise maybe sort it outside of this function?